desktop/libs/notebook/src/notebook/connectors/spark_shell.py

```diff
@@ -160,12 +160,16 @@ def create_session(self, lang='scala', properties=None):
       if USE_DEFAULT_CONFIGURATION.get():
         config = DefaultConfiguration.objects.get_configuration_for_user(app='spark', user=self.user)
 
+      # Populate props with default configuration
+      properties = self.get_properties()
+      props = dict([(p['name'], p['value']) for p in properties]) if properties is not None else {}
```

Comment:
It starts to be a lot of code duplication. Move out 'props = dict([(p['name'], p['value']) for p in properties]) if properties is not None else {}' and reuse it?

Reply:
My logic was:
1) If variable `properties` is passed to a function `create_session` as argument
 then no merge with defaults if required.

May be it's necessary to merger with default properties in this case too?

2) If variable `properties` is None:
2.1.  populate dict `props` with default properties
2.2. replace element in dict props with custom values


If we can always merger with defaults, code can be:
```python
# Populate with default properties
props = dict([(p['name'], p['value']) for p in self.get_properties()])

if not properties:
  config = None
  if USE_DEFAULT_CONFIGURATION.get():
    config = DefaultConfiguration.objects.get_configuration_for_user(app='spark', user=self.user)

  if config is not None:
    properties = config.properties_list
    
props.update(dict([(p['name'], p['value']) for p in properties])) if properties is not None else {}
```

Is It better?

Reply:
Yes, leaner!

And I looked a bit deeper and the original code was indeed super confusing :)

Actually, maybe moving
```
        if USE_DEFAULT_CONFIGURATION.get():
        config = DefaultConfiguration.objects.get_configuration_for_user(app='spark', user=self.user)
```
into self.get_properties()? 
(to manage all the defaults there)
        
And then we can keep it simple in create_session() by just overriding them:

```
# Populate with default properties
props = dict([(p['name'], p['value']) for p in self.get_properties()])

if properties is not None:    
  props.update(dict([(p['name'], p['value']) for p in properties]))
```